src/frequenz/sdk/timeseries/_ringbuffer.py

```diff
@@ -220,14 +220,17 @@ def window(
 
         # Requested window wraps around the ends
         if start_index >= end_index:
-            window: Any = self._buffer[start_index:]
-
             if end_index > 0:
                 if isinstance(self._buffer, list):
-                    window += self._buffer[0:end_index]
+                    return self._buffer[start_index:] + self._buffer[0:end_index]
+                elif isinstance(self._buffer, np.ndarray):
+                    return np.concatenate(
+                        (self._buffer[start_index:], self._buffer[0:end_index])
+                    )
                 else:
-                    window = np.concatenate((window, self._buffer[0:end_index]))
-            return window
+                    raise TypeError(
+                        "Bad type, something seems to have went wrong with type checking"
+                    )
```

Comment:
I think this change will need to update the docstring to add `TypeError` in the `Raises` session

Reply:
I think this is actually a good reason to make it an `assert`, I didn't want to use `assert False`, but it doesn't make sense to document an error that will only happen if type checking is not performed.

Reply:
Right, I see your point. Though I don't know if the nox-docstrings checks will complain about it.

Reply:
No, it doesn't.